tee-worker/enclave-runtime/src/initialization/mod.rs

```diff
@@ -210,20 +213,32 @@ pub(crate) fn init_enclave(
 
 	let data_provider_config = GLOBAL_DATA_PROVIDER_CONFIG.get()?;
 	let getter_executor = Arc::new(EnclaveGetterExecutor::new(state_observer));
-	let io_handler = public_api_rpc_handler(
+
+	let mut io_handler = IoHandler::new();
+
+	add_common_api(
+		&mut io_handler,
 		top_pool_author,
 		getter_executor,
 		shielding_key_repository,
 		ocall_api.clone(),
 		Some(state_handler),
 		data_provider_config,
 	);
+
+	#[cfg(feature = "sidechain")]
+	{
```

Comment:
In general, it looks good to me. I am just wondering: is it this block the fundamental change?

Reply:
The main change of this PR is in this file:
`tee-worker/service/src/worker.rs`